### PR TITLE
Fixes 127: fix version filter using group clause

### PR DESCRIPTION
**Pull request by jlsherrill**
Does the following:

* version filter on repo list api was improperly constructing a query, it was doing:
   org_id = X AND version = 7 OR version = 9   which would improperly return all
   repos with version = 9.  The new query should properly be structured like:
   org_id = X AND (version = 7 OR version = 9)
* Stops dropping all tables as part of each test, this speeds up the tests greatly when
   there is seeded data
* Starts using random Org ids for tests

reproducer steps:

in a fresh database:

Create a repo in org 3 with el7 & el9
```
$ curl -X POST http://localhost:8000/api/content_sources/v1/repositories/   -H "`scripts/header.sh 3 3`"  -d '{"url": "http://foo.com/bar", "distribution_versions": ["el7", "el9"], "name": "foo"}'  -H "Content-Type: application/json" 
```

Create a repo in org 1 with el7 and el9
```
curl -X POST http://localhost:8000/api/content_sources/v1/repositories/   -H "`scripts/header.sh 1 1`"  -d '{"url": "http://foo.com/bar", "distribution_versions": ["el7", "el9"], "name": "foo"}'  -H "Content-Type: application/json"
```
query repos in org 1 with just el7 or el9, should only return a single entry, but we get two:

```
curl -X GET "http://localhost:8000/api/content_sources/v1/repositories/?version=el7,el9" -H "`scripts/header.sh 1 1`" 
{
   "data":[
      {
         "uuid":"6092ce16-b2ec-4d09-a4c4-c1eb35d9dea6",
         "name":"foo",
         "url":"http://foo.com/bar",
         "distribution_versions":[
            "el7",
            "el9"
         ],
         "distribution_arch":"",
         "account_id":"3",
         "org_id":"3"
      },
      {
         "uuid":"c86a011a-d688-4f9f-abe6-7af33d8ca20c",
         "name":"foo",
         "url":"http://foo.com/bar",
         "distribution_versions":[
            "el7",
            "el9"
         ],
         "distribution_arch":"",
         "account_id":"1",
         "org_id":"1"
      }
   ],
   "meta":{
      "limit":100,
      "offset":0,
      "count":2
   },
   "links":{
      "first":"/api/content_sources/v1/repositories/?limit=100\u0026offset=0\u0026version=el7,el9",
      "last":"/api/content_sources/v1/repositories/?limit=100\u0026offset=0\u0026version=el7,el9"
   }
}
```

### CODE REVIEW

**Comment by jlsherrill** [comment]
Note this is built ontop of https://github.com/content-services/content-sources-backend/pull/45   will rebase once that goes in

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMSCONTENT-127

File: pkg/dao/repository.go [hunk 1]
```diff
@@ -100,10 +100,11 @@ func (r repositoryDaoImpl) List(
 
 	if filterData.Version != "" {
 		versions := strings.Split(filterData.Version, ",")
-		filteredDB = filteredDB.Where("? = any (versions)", versions[0])
+		orGroup := r.db.Where("? = any (versions)", versions[0])
 		for i := 1; i < len(versions); i++ {
-			filteredDB.Or("? = any (versions)", versions[i])
```
Comment (Andrewgdewar) [inline pkg/dao/repository.go:105]: would: 
filteredDB = filteredDB.Or("? = any (versions)", versions[i]) 
not work here?

Comment (Andrewgdewar) [reply pkg/dao/repository.go:105]: I'm happy to keep it with your method, just curious.

**Comment by jlsherrill** [comment]
Built and pushed image for 4acf7e8f4c8a04dc0b8b6bd2ff120a0c2096c1fe

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/14751236

**Comment by jlsherrill** [comment]
Built and pushed image for f215900a1f98d099859b5bfdb3c8e517aeac7668

**Review by rverdile — APPROVED** [review]
lgtm!

**Review by swadeley — APPROVED** [review]
ACK

**Pull request merged**